### PR TITLE
core: parse socket port to extract_first_word

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/core/load-fragment.c [hunk 1]
```diff
@@ -1591,21 +1591,30 @@ int config_parse_service_sockets(
...
 
-                t = strndup(word, l);
-                if (!t)
+                r = extract_first_word(&cur, &word, NULL, 0);
+                if (r == 0)
+                        break;
+                if (r == -ENOMEM)
                         return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Trailing garbage in sockets, ignoring: %s", prev);
```
Comment (zonque) [inline src/core/load-fragment.c]: That's the only location where `prev` is really used, correct? Why not just log `rvalue`, so you can avoid `prev` all together?

Comment (ssahani) [inline src/core/load-fragment.c]: I think 'prev'  will print the exact garbage value .  'rvalue'  will the whole line .

Comment (zonque) [inline src/core/load-fragment.c]: Well, no, `cur` is the current pointer, and extracting a word from that failed if you get here. Hence, logging `cur` is sufficient. No need to preserve the former pointer.

Comment (ssahani) [inline src/core/load-fragment.c]: any specific reason it's been used in 
config_parse_bounding_set and config_parse_namespace_path_strv ?

Comment (zonque) [inline src/core/load-fragment.c]: That seems unnecessary to me, really. I still think it's totally fine to just log `rvalue` in that case.

Comment (ssahani) [inline src/core/load-fragment.c]: Well yes printing the whole is nice .

Comment (evverx) [inline src/core/load-fragment.c]: @zonque , I added `prev`: https://github.com/systemd/systemd/pull/1647#issue-112887252

```
[/etc/systemd/system/a.service:3] Not an absolute path, ignoring: /boot a /usr b -what /b�t hm'hola
```

What was ignored here?:)
The whole line is unusable for debugging.

Comment (evverx) [inline src/core/load-fragment.c]: > Well, no, cur is the current pointer, and extracting a word from that failed if you get here. Hence, logging cur is sufficient. No need to preserve the former pointer.

`extract_first_word` [changes](https://github.com/systemd/systemd/blob/b11d6a7bed4d867fb9f6ff4e7eb4ab20fcdc9301/src/basic/extract-word.c#L170) `cur`. so, I can have `-EINVAL` and shifted `cur`.
[Example](https://github.com/systemd/systemd/blob/b11d6a7bed4d867fb9f6ff4e7eb4ab20fcdc9301/src/test/test-extract-word.c#L186):

```
        p = original = "\"foo\\";
        assert_se(extract_first_word(&p, &t, NULL, 0) == -EINVAL);
        assert_se(p == original + 5);
```

**Pull request merged**